bundles/org.openhab.binding.linktap/src/main/java/org/openhab/binding/linktap/protocol/http/NotTapLinkGatewayException.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.linktap.protocol.http;
+
+import java.io.Serial;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * The {@link NotTapLinkGatewayException} should be thrown when the endpoint being communicated with
+ * does not appear to be a Tap Link Gateway device.
+ *
+ * @author David Goodyear - Initial contribution
+ */
+@NonNullByDefault
+public class NotTapLinkGatewayException extends Exception {
+    @Serial
+    private static final long serialVersionUID = -7786449325604153487L;
+
+    public NotTapLinkGatewayException() {
+        super();
+    }
+
+    public NotTapLinkGatewayException(final String message) {
+        super(message);
+    }
+
+    public NotTapLinkGatewayException(final Throwable cause) {
+        super(cause);
+    }
+
+    public NotTapLinkGatewayException(final String message, final Throwable cause) {
+        super(message, cause);
+    }
+
+    public static final String HEADERS_MISSING = "Missing header markers";
+    public static final String MISSING_API_TITLE = "Not a LinkTap API response";
+    public static final String MISSING_SERVER_TITLE = "Not a LinkTap response";
+    public static final String UNEXPECTED_STATUS_CODE = "Unexpected status code response";
+    public static final String UNEXPECTED_HTTPS = "Unexpected protocol";
```

Comment:
Did you consider to use the i18n provider to support translated error messages?
For reference:https://www.openhab.org/docs/developer/utils/i18n.html#i18n-text-provider-api

Reply:
Everything that updates the status, or prints a warning to the console I believe I have added to the i18 system. Ironically the above isn't used yet in any messages from my code scans.

Reply:
They are used in the webserverapi.java

Reply:
Yeah I think they get re-wrapped when caught to more useful errors. I have worked through and mapped to i18n as there may have been a case that output the message directly. Now at least i18n errors can be presented if required.